tests/Integration/DataAccess/DoctrineDonationRepositoryTest.php

```diff
@@ -112,4 +112,14 @@ public function testWhenDoctrineThrowsException_domainExceptionIsThrown() {
 		$repository->getDonationById( 42 );
 	}
 
+	public function testWhenPersistingDeletedApplication_exceptionIsThrown() {
```

Comment:
Could you rename this to `testWhenPersistingNonexistentApplication_exceptionIsThrown`? I was momentarily confused because I look at the test code and did non know where you "deleted" the Application.